tasks/nfs-clients.yml

```diff
@@ -7,10 +7,15 @@
   file:
     path: "{{ nfs_client_mnt_point }}"
     state: directory
+- name: unmount existing nfs mounts if not the desired one
+  mount:
+    path: "/mnt/nfs"
+    state: absent
+  when: nfs_client_mnt_src not in nfs_current_mounts
 - name: mount the filesystem
   mount:
-    path: "{{ nfs_client_mnt_point}}"
-    src: "{{ nfs_server }}:{{ nfs_export }}"
+    path: "{{ nfs_client_mnt_point }}"
+    src: "{{ nfs_client_mnt_src }}"
```

Comment:
why doesn't this detect the change in mount, any ideas?

Reply:
Good question, having had a look at the code, it invokes a `mount -o remount` which doesn't remount the specified source path as expected, only seems to update the fstab file... unmount and remount gets around the issue though. I haven't seen this with block devices, but only with NFS.